libs/extensions/src/settings/settings.service.ts

```diff
@@ -171,24 +166,56 @@ export class SettingsService {
       throw new Error('Invalid data structure');
     }
 
-    const matchKeysWithRequiredFields = Object.keys(value).every((key) =>
-      requiredFields.includes(key),
-    );
+    const requiredFieldsArray: string[] = requiredFields
```

Comment:
can you wrap this process in a function. I guess there is same code in create side too.
Cleanup the code

Reply:
@saroj-rumsan

Reply:
ok